mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb

```diff
@@ -1748,6 +1784,30 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 		return fmt.Errorf("Instance had unexpected number of network interfaces: %d", len(instance.NetworkInterfaces))
 	}
 
+	<% unless version == 'ga' -%>	
+	updateSecurityPolicy := false
+	for i := 0; i < len(instance.NetworkInterfaces); i++ {
+		prefix := fmt.Sprintf("network_interface.%d", i)
+		// check if sec policy has been changed
+		// check if access config has been changed because it may be deleted and needs to be re-created.
+		if d.HasChange(prefix+".security_policy") || d.HasChange(prefix+".access_config") || d.HasChange(prefix+".ipv6_access_config") {
```

Comment:
If the security_policy doesn't change, only access_config changes, updateSecurityPolicy is still set to true?

Reply:
What is the reason to check if access config has been changed? Sorry I don't get it.

Reply:
The security_policy info is kept inside access_config. Once the access_config is removed, the security_policy will be gone requiring to be applied again. In order to update access_config, the code is deleting and re-creating again. I will mark the logic in this file which deals with that.